client/src/components/organisms/RequestGroupForm.tsx

```diff
@@ -0,0 +1,351 @@
+  const [imageError, setImageError] = useState("")
+  const [requestTypesError, setRequestTypesError] = useState("")
+
+  const images = ["https://source.unsplash.com/RcgiSN482VI",
+    "https://source.unsplash.com/7ydep8OEvbc",
+    "https://source.unsplash.com/0hiUWSi7jvs",
+    "https://source.unsplash.com/uWVWQ8gF8PE",
+    "https://source.unsplash.com/ZQ_0jk66-1E",
+    "https://source.unsplash.com/7EWCIzU4TIg",
+    "https://source.unsplash.com/FOrNjVWdUzM",
+  ]
+
+
+  const requestGroupQuery = gql`
+  query FetchRequestGroup($id: ID!) {
+    requestGroup(id: $id) {
+      id
+      name
+      description
+      image
+      requestTypes {
+        id
+        name
+        deleted
+      }
+    }
+  }`
+
+  if (props.requestGroupId) {
+    useQuery(requestGroupQuery, {
+      variables: { id: props.requestGroupId, },
+      onCompleted: (data: { requestGroup: RequestGroup }) => {
+        const res: RequestGroup = JSON.parse(JSON.stringify(data.requestGroup)); // deep-copy since data object is frozen
+
+        setInitialRequestGroup(res)
+        setName(res.name ? res.name : "")
+        setDescription(res.description ? res.description : "")
+        setImage(res.image ? res.image : "")
+
+        res.requestTypes?.forEach((requestType) => {
+          if (requestType.name && requestType._id) {
+            setRequestTypesToIdMap(requestTypesToIdMap.set(requestType.name, requestType._id))
+          }
+        })
+
+        setRequestTypeNames(
+          res.requestTypes ?
+            res.requestTypes
+              .filter((requestType) => requestType.name)
+              .map((requestType) => requestType.name as string)
+            : []
+        )
+      }
+    });
+  }
+
+  const getNameError = (name: string): string => {
+    if (name.length === 0) {
+      return "Please enter a group name";
+    }
+    if (name.length > 20) {
+      return "Group name cannot exceed 20 characters";
+    }
+    if (props.requestGroups.find((requestGroup) => requestGroup.name === name && requestGroup._id !== props.requestGroupId)) {
+      return "There is already a group with this name";
+    }
+
+    return "";
+  }
+
+  const getInputRequestTypeNameError = (inputRequestTypeName: string): string => {
+    if (requestTypeNames.find(requestTypeName => requestTypeName === inputRequestTypeName)) {
+      return "There is already a type with this name";
+    }
+    else if (inputRequestTypeName.length > 40) {
+      return "Type name cannot exceed 40 characters";
+    }
+    return "";
+  }
+
+  const getRequestTypeNamesError = (requestTypes: Array<string>): string => {
+    if (requestTypes.length === 0) {
+
+      return "Please create at least 1 type";
+    }
+    return "";
+  }
+
+  const getDescriptionError = (description: string) => {
+    if (!description) {
+      return "Please enter a description";
+    }
+
+    const editorState = EditorState.createWithContent(convertFromRaw(JSON.parse(description)))
```

Comment:
Please instead add an onEmpty prop or some other optional prop for the RichTextField. This breaks encapsulation of the component and imo is undesired

Reply:
will do